### ISSUE TITLE
Possible resume bug in training manager

### ISSUE THREAD

**Issue by msheller** [CBICA/GaNDLF#430]
https://github.com/CBICA/GaNDLF/blob/f6d26038a9d6203941748e43a593663594a3b0c0/GANDLF/training_manager.py#L28

Hey guys,

Any chance this should be "or not resume"? It reads like it will create a new parameters.pkl file if resume is true, which doesn't match my expectation.

**Comment by Geeks-Sid** [comment]
Hi @msheller , This was a feature because sometimes while resuming training, we might want to change a few hyperparameters (such as add augmentation, change the learning rate for fine tuning/transfer learning) which is why we have written it in such a way.

@sarthakpati and I have discussed about adding a feature later on in which we will allow users to change other parameters but not the model parameters themselves (such as number of filters, output channels etc) which will allow us to detect if the user is trying to do an invalid operation (loading the weights to a different model).

But for now, This is needed to create new config files even during reset or resume.

Would be happy to discuss.

**Comment by msheller** [comment]
@Geeks-Sid So have the model weights already been loaded into the parameters dictionary by this point?

**Comment by Geeks-Sid** [comment]
@msheller  Could you clarify what you mean by model weights being loaded into parameters dictionary?

Are we talking about the parameters (parameters such as model, num_classes, optimizer etc) that are given by the user through the config file or the parameters dictionary used for loading model weights (This dictionary is created automatically for a model) into the model's parameter dictionary?

**Comment by sarthakpati** [comment]
> @Geeks-Sid So have the model weights already been loaded into the parameters dictionary by this point?

I think the model weights get loaded after that point in https://github.com/CBICA/GaNDLF/blob/f6d26038a9d6203941748e43a593663594a3b0c0/GANDLF/compute/training_loop.py#L297

**Comment by msheller** [comment]
@sarthakpati @Geeks-Sid There it is :) Thanks, guys! I'll close this up.
